sympy/combinatorics/homomorphisms.py

```diff
@@ -417,3 +420,125 @@ def block_homomorphism(group, blocks):
+    >>> from sympy.combinatorics.named_groups import DihedralGroup, AlternatingGroup
+
+    >>> D = DihedralGroup(8)
+    >>> p = Permutation(0, 1, 2, 3, 4, 5, 6, 7)
+    >>> P = PermutationGroup(p)
+    >>> group_isomorphism(D, P, isomorphism=True)
+    (False, None)
+
+    >>> F, a, b = free_group("a, b")
+    >>> G = FpGroup(F, [a**3, b**3, (a*b)**2])
+    >>> H = AlternatingGroup(4)
+    >>> (check, T) = group_isomorphism(G, H, isomorphism=True)
+    >>> check
+    True
+    >>> T(b*a*b**-1*a**-1*b**-1)
+    (0 2 3)
+
+    '''
+    if not isinstance(G, (PermutationGroup, FpGroup)):
+        raise TypeError("The group must be a PermutationGroup or a finite FpGroup")
+    if not isinstance(H, (PermutationGroup, FpGroup)):
+        raise TypeError("The group must be a PermutationGroup or a finite FpGroup")
+
+    if isinstance(G, FpGroup) and isinstance(H, FpGroup):
+        G = simplify_presentation(G)
+        H = simplify_presentation(H)
+        # Two infinite FpGroups with the same generators are isomorphic
+        # when the relators are same but are ordered differently.
+        if G.generators == H.generators and (G.relators).sort() == (H.relators).sort():
+            if not isomorphism:
+                return True
+            return (True, homomorphism(G, H, G.generators, H.generators))
+
+    # `_G` and `_H` are the permutation groups isomorphic to
+    # `G` and `H` respectively, if they are `FpGroups`
+    _G = G
+    group_isomorphism = None
+    _H = H
+    h_isomorphism = None
+    g_order = G.order()
+    h_order = H.order()
+
+    if isinstance(G, FpGroup):
+        if g_order == S.Infinity:
+            raise NotImplementedError("Isomorphism methods are not implemented for infinite groups.")
+        _G, g_isomorphism = G._to_perm_group()
+    if isinstance(H, FpGroup):
+        if h_order == S.Infinity:
+            raise NotImplementedError("Isomorphism methods are not implemented for infinite groups.")
+        _H, h_isomorphism = H._to_perm_group()
+
+    if (g_order != h_order) and (G.is_abelian != H.is_abelian):
+        if not isomorphism:
+            return False
+        return (False, None)
+
+    if not isomorphism:
+        n = g_order
+        if (g_order == h_order) and (igcd(n, totient(n))) == 1:
+            return True
+
+    # Match the generators of `_G` with the subsets of `_H`
+    gens = _G.generators
+    for subset in itertools.permutations(_H, len(gens)):
+        gens = list(gens)
+        images = list(subset)
+        images.extend([_H.identity]*(len(_G.generators)-len(images)))
+        gens.extend([g for g in _G.generators if g not in gens])
+        _images = dict(zip(gens,images))
+        if _check_homomorphism(_G, _H, _images):
+            images = []
+            gens = []
+            if isinstance(H, FpGroup):
+                for s in subset:
+                    images.append(h_isomorphism.invert(s))
```

Comment:
`images = h_isomorphism.invert(subset)` should do it, provided `subset` is a list.